iotilecore/iotile/core/utilities/async_tools/operation_manager.py

```diff
@@ -254,11 +271,75 @@ def wait_for(self, timeout=None, **kwargs):
             raise ArgumentError("You must specify at least one message field to wait on")
 
         spec = MessageSpec(**kwargs)
-        future = self._add_waiter(spec)
+        future = self._add_waiter(spec, pause=pause)
         future.add_done_callback(lambda x: self._remove_waiter(spec, future))
 
+        if unpause:
+            self.unpause()
+
         return asyncio.wait_for(future, timeout=timeout)
 
+    async def gather_until(self, gather_specs, until, timeout=None, *, pause=False, unpause=False):
```

Comment:
Looking at this function alone, I can't easily tell what the arguments coming in are supposed to be (`gather_specs` and `until`)

Reply:
Docstrings updated in latest commits.